components/apimgt/org.wso2.carbon.apimgt.persistence.mongodb/src/main/java/org/wso2/carbon/apimgt/persistence/mongodb/MongoDBPersistenceImpl.java

```diff
@@ -0,0 +1,813 @@
+            throw new DocumentationPersistenceException("Failed to delete documentation. Cannot find " + docId + "in " +
+                    "mongodb for api " + apiId);
+        }
+        return apiDocumentation;
+    }
+
+    private void handleFileTypeContent(Organization org, String apiId, String docId, DocumentContent content)
+            throws DocumentationPersistenceException {
+        MongoDatabase database = MongoDBConnectionUtil.getDatabase();
+        MongoCollection<MongoDBPublisherAPI> collection = getPublisherCollection(org.getName());
+        GridFSBucket gridFSFilesBucket = GridFSBuckets.create(database, org.getName());
+        ResourceFile resourceFile = content.getResourceFile();
+        InputStream inputStream = resourceFile.getContent();
+        GridFSUploadOptions options = new GridFSUploadOptions().chunkSizeBytes(358400);
+        String textContent = null;
+        String contentType = resourceFile.getContentType();
+        try {
+            File file = MongoDBUtil.writeStream(inputStream, resourceFile.getName());
+            InputStream extractStream = MongoDBUtil.readStream(file, resourceFile.getName());
+            if (contentType.equalsIgnoreCase(APIConstants.DOCUMENTATION_PDF_CONTENT_TYPE)) {
+                textContent = MongoDBUtil.extractPDFText(extractStream);
+            }
+            if (contentType.equalsIgnoreCase(APIConstants.DOCUMENTATION_DOC_CONTENT_TYPE)) {
+                textContent = MongoDBUtil.extractDocText(extractStream);
+            }
+            if (contentType.equalsIgnoreCase(APIConstants.DOCUMENTATION_DOCX_CONTENT_TYPE)) {
+                textContent = MongoDBUtil.extractDocXText(extractStream);
+            }
+            if (contentType.equalsIgnoreCase(APIConstants.DOCUMENTATION_TXT_CONTENT_TYPE)) {
+                textContent = MongoDBUtil.extractPlainText(extractStream);
+            }
+
+            InputStream gridFsStream = MongoDBUtil.readStream(file, resourceFile.getName());
+            ObjectId gridFsReference =
+                    gridFSFilesBucket.uploadFromStream(resourceFile.getName(), gridFsStream, options);
+
+            collection.updateOne(
+                    and(
+                            eq("_id", new ObjectId(apiId)),
+                            eq("documentationList.docId", new ObjectId(docId))
+                    ),
+                    combine(
+                            set("documentationList.$.gridFsReference", gridFsReference),
+                            set("documentationList.$.contentType", contentType),
+                            set("documentationList.$.textContent", textContent)
+                    )
+            );
+        } catch (IOException | PersistenceException e) {
+            log.error("Error when extracting text content from file ", e);
+            throw new DocumentationPersistenceException("Failed to documentation content for " + docId +
+                    " in mongodb, for api " + apiId);
```

Comment:
Do not log and throw. Lets throw only. Lets include "e" in the thrown exception.
```suggestion
            throw new DocumentationPersistenceException("Failed to extract documentation content for " + docId +
                    " in mongodb, for api " + apiId, e);
```